umi_tools/umi_methods.py

```diff
@@ -30,13 +30,28 @@
     from _dedup_umi import edit_distance
 
 
-def get_umi(read, sep="_"):
-    try:
-        return read.qname.split(sep)[-1].encode('utf-8')
-    except IndexError:
-        raise ValueError("Could not extract UMI from read, please"
-                         "check UMI is encoded in the read name"
-                         "following the final '_' or run with --ignore-umi")
+def get_umi(read, sep="_", tag='RX', method='read_id'):
+    ''' extract the umi from either:
+    - read id using specified separator
+    - specified tag'''
+
+    if method == 'read_id':
+        try:
+            return read.qname.split(sep)[-1].encode('utf-8')
+        except IndexError:
+            raise ValueError(
+                "Could not extract UMI from the read ID, please"
+                "check UMI is encoded in the read name")
+
+    elif method == 'tag':
+        try:
+            return read.get_tag(tag).encode('utf-8')
+        except IndexError:
+            raise ValueError(
+                "Could not extract UMI from the read ID, please"
```

Comment:
Error message should say "Could not extract UMI from the read tag, please [...]"